src/c++/perf_analyzer/load_manager.h

```diff
@@ -86,9 +86,12 @@ class LoadManager {
     }
     // The backend to communicate with the server
     std::unique_ptr<cb::ClientBackend> infer_backend_;
-    // The vector of pointers to InferInput objects to be
-    // used for inference request.
+    // The vector of pointers to InferInput objects for each input in the model
+    // config.
     std::vector<cb::InferInput*> inputs_;
+    // The vector of pointers to InferInput objects for each input in the model
+    // config except optional inputs that have no user data provided.
+    std::vector<cb::InferInput*> final_inputs_;
```

Comment:
I'm not sure why we need both `final_inputs_` and `inputs_`. Would it be possible for `inputs_` to contain only a list of inputs that the client is going to send to the server? If there are unspecified optional inputs, it will contain fewer elements.

Reply:
This design was mainly @tanmayv25's idea. The problem with `inputs_` containing "only what it needs to" is that inference is split up across two functions:

* `PrepareInfer()`, which is run for the first inference, which loops through `ModelParser::Inputs`, and fills `inputs_` with all of them, and `final_inputs_` with the ones with input data.
* `UpdateInputs()`, which is for all subsequent inferences, which loops through the `inputs_` created in `PrepareInfer()` and clears/refills `final_inputs_` with the ones with input data.

Because of this design, we need a data structure that remembers "all _possible_ inputs". That's what `inputs_` does for us currently. Then `final_inputs_` is the "place to hold data-provided inputs per inference".